org.eclipse.xtext/src/org/eclipse/xtext/formatting2/internal/MultilineCommentReplacer.java

```diff
@@ -36,8 +36,12 @@ public void configureWhitespace(WhitespaceReplacer leading, WhitespaceReplacer t
 			enforceNewLine(leading);
 			enforceNewLine(trailing);
 		} else {
-			enforceSingleSpace(leading);
-			enforceSingleSpace(trailing);
+			if (!leading.getRegion().isMultiline()) {
+				enforceSingleSpace(leading);
```

Comment:
Not sure about this guard here.

Reply:
I've added MLSLOnSecondLine to show why this is needed